azurelinuxagent/common/exception.py

```diff
@@ -184,12 +184,6 @@ class ProtocolNotFoundError(ProtocolError):
     """
 
 
-class IncompleteGoalStateError(ProtocolError):
```

Comment:
I did not see the need to have a separate exception for this error so I changed the code to use ProtocolError.

This requires a change in DCR, which looks for IncompleteGoalStateError. I am preparing that change and will submit the PR on the tux repo.